k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java

```diff
@@ -2012,7 +2012,16 @@ private View getFooterView(ViewGroup parent) {
 
     private void updateFooterView() {
         if (!mSearch.isManualSearch() && mCurrentFolder != null && mAccount != null) {
-            if (mCurrentFolder.loading) {
+            int msg=100;
+            try {
+                msg=mCurrentFolder.folder.getMessageCount();
+            } catch (MessagingException e) {
+                e.printStackTrace();
+            }
```

Comment:
We don't want to be logging an exception like this if it fails.

However a more interesting question is why the method can even throw this.

It turns out that a LocalFolder, POP3 Folder and IMAP folder just return a value. So they will never throw this exception and it'll be a fairly quick call.

However WebDAV appears to not cache this value at all. It does a server query every time.

I think we should fix this behaviour in WebDAV first and then there'll be no need to handle this exception here and this whole logic can be simplified.